src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
+    mFields["organizationName"] = "Lv-425.C++";
+    mFields["style"] = "WHITE";
+    mFields["editorFontName"] = "Consolas";
+    mFields["editorFontSize"] = "12";
+    mFields["defaultUdpPortNumber"] = "36108";
+    mFields["defaultTcpPortNumber"] = "32807";
+    mFields["userName"] = "unnamed user";
+    mFields["cppExtentions"] = ".c;.cpp;.h;.hpp;.json;.txt";
+    mCppExtentionsList << ".c" << ".cpp" << ".h" << ".hpp" << ".json" << ".txt";
+
+}
+
+void StoreConf::restoreConFile()
+{
+    checkConfDir();
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (!file.exists()) // userName.json file not exist. keep all fields with default values,
+    {                   // create userName.json with these values
         writeJson();
         saveData();
-    } else {
+    }
+    else
+    {
         readJson();
-        parseJson();
+        if (mReadStatus)
+        {
+            parseJson();
+        }
+        else
+        {                        //  file corrupted
+            writeJson();        //      rewrite with default values
+        }
         saveData();
     }
 }
 
-QString StoreConf::getPathToConFile()
+void StoreConf::saveConFile()
 {
-    QString currentPath = QDir::currentPath();
-    currentPath += "/conf.json";
-    QFile file(currentPath);
-    if(file.exists())
-        return "conf.json";
-    else
-        return "";
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (file.exists())
+    {
+        writeJson("finish");
+    }
+}
+
+void StoreConf::getPathToConFile()
+{
+    mPathToConFile = QDir::currentPath();
+    mPathToConFile += "/conf/";
+    mPathToConFile += mConFile;
+}
+
+void StoreConf::checkConfDir(QString str)
+{
+    if (str == "exe")
+    {
+        QString pathToConDir = QDir::currentPath();
+        pathToConDir += "/conf";
+        if (!QDir(pathToConDir).exists())
+            QDir().mkdir(pathToConDir);
+    }
 }
 
-void StoreConf::writeJson()
+void StoreConf::writeJson(QString mode)
 {
     QJsonObject root_obj;
-    root_obj.insert("organizationName", organizationName);
-    root_obj.insert("applicationVersion", applicationVersion);
-    root_obj.insert("applicationName", applicationName);
-    root_obj.insert("analizerStyle", analizerStyle);
-    root_obj.insert("analizerFontSize", analizerFontSize);
-    root_obj.insert("analizerFontName", analizerFontName);
-    root_obj.insert("cppExtentions", cppExtentions);
+
+    QSettings s;
+    QMap<QString, QString>::const_iterator it = mFields.constBegin();
+    while (it != mFields.constEnd())
+    {
+        if (mode == "start")
+        {                           // copy value from QMap to json
+            root_obj.insert(it.key(), it.value());
+        }
+        else
+        {                           // copy value from QSettings to json
+            if (s.contains(it.key()))
+                root_obj.insert(it.key(), s.value(it.key()).toString());
+        }
```

Comment:
Why are data written to both QSettings repository & json file?

Reply:
all values from configuration file application load during starting and save values at the quiting.
during current session all changes (can be many times) reflecting in QSettings as global available place.